framework/deproxy_client.py

```diff
@@ -251,7 +259,7 @@ def wait_for_response(self, timeout=5):
             return False
 
         t0 = time.time()
-        while len(self.responses) < self.valid_req_num:
+        while len(self.responses) < self.valid_req_num or self.valid_req_num == 0:
```

Comment:
`or self.valid_req_num == 0` is not necessary if `make_requests` will work correctly.